### DISCUSSION TITLE
Prisma migrations

### DISCUSSION THREAD

**Discussion by adamalfredsson** [planetscale/discussion#415] [Q&A] [answered]
I'm trying PlanetScale for a new project, but I'm still struggling to understand the concept of automatic migrations. I've watched [this video](https://www.youtube.com/watch?v=iaHt5_hg44c) and read [this documentation](https://planetscale.com/docs/tutorials/automatic-prisma-migrations).

In the video, Norman demonstrates a deploy request by adding `createdAt` and `updatedAt` columns to a model in his schema. Now I'm trying the exact same thing in my user schema:

```
 CREATE TABLE `User` (
  `id` varchar(191) NOT NULL,
+ `createdAt` datetime(3) NOT NULL DEFAULT current_timestamp(3),
+ `updatedAt` datetime(3) NOT NULL,
  `name` varchar(191),
  `email` varchar(191),
  `emailVerified` datetime(3),
  `image` varchar(191),
  PRIMARY KEY (`id`),
  UNIQUE KEY `User_email_key` (`email`)
 ) ENGINE InnoDB,
  CHARSET utf8mb4,
  COLLATE utf8mb4_unicode_ci;
```

But once I apply my deploy request I get this error:

> Field 'updatedAt' doesn't have a default value (errno 1364) (sqlstate HY000) during query: insert into User(id,`name`,email,emailVerified,image)

I just feel really dumb and frustrated at this point, since it's probably a very trivial issue. I really don't feel like I've found good pointers on how to solve this. What am I missing?

**Comment by jgreet** [comment, marked as answer]
The fix is either to remove the `NOT NULL` constraint on your new `updatedAt` field, or provide a default value.

In an empty table this migration would run fine, but if you have existing rows in the table [the online schema migration process](https://planetscale.com/docs/learn/how-online-schema-change-tools-work#the-general-logic-behind-online-schema-change-tools) won't be able to insert them in to new version of the `User` table due to the error in your message.

Hope that clears it up. Let us know if there's anything else we can help with here.
